test/distributed/fsdp/test_fsdp_mixed_precision.py

```diff
@@ -860,9 +883,14 @@ def test_full_precision_in_eval_buffers(self):
             # model.eval() + forward pass should make the buffers in full prec again
             fsdp_model.eval()
             fsdp_model((inp, self, fsdp_model, mp_config, torch.float32))
-
+            # TODO: this test would be more robust if the buffer dtype was
+            # validated in the nn.Module pre-forward hook to ensure that the
+            # buffer computation takes place in the right precision.
             for buf in fsdp_model.buffers():
-                self.assertEqual(_BUFFER_ORIG_DTYPE, buf.dtype)
+                if full_prec_in_eval:
```

Comment:
nit: You do `expected_dtype = ... if full_prec_in_eval else ...` above. You could do it here too for slight improvement in readability.